### PR TITLE
Adding Perf API + fixing typo

### PR DESCRIPTION
**Pull request by robisonjoel**
* Adding netapp/perf.go
* changing DiagnosisAlertPagesResposne to DiagnosisAlertPagesResponse

### CODE REVIEW

File: netapp/perf.go [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+	return &response, rawResponse, err
+}
+
+type PerfObjectInstanceListInfoHandler func(PerfObjectInstanceListInfoPageResponse) (shouldContinue bool)
+
+func (p *Perf) PerfObjectInstanceGetAllInfo(options *PerfObjectInstanceListInfoIterParams, fn PerfObjectInstanceListInfoHandler) {
+
+	requestOptions := options
+
+	for shouldContinue := true; shouldContinue; {
```
Comment (pyama86) [inline netapp/perf.go:141]: I think that this way of writing is simple
```
for  {
[snip]
  if nexttag != "" && handlerResponse {
    break
  }
}
```

**Review by pyama86 — CHANGES_REQUESTED** [review]
Thank you for PR. I commented one.

File: netapp/perf.go (hunk 1 above)
Comment (robisonjoel) [reply netapp/perf.go:141]: you have me at a loss, im not sure what you mean here.

Comment (pyama86) [reply netapp/perf.go:141]: I'm telling you that you should not need `shouldContinue`.

Comment (robisonjoel) [reply netapp/perf.go:141]: Trying to keep the codebase consistent - This was patterned after:
https://github.com/pepabo/go-netapp/blob/master/netapp/aggregate.go#L52
https://github.com/pepabo/go-netapp/pull/5/files/aa590074789adebcf3e970ef9bded90fb89dfb3b#diff-942f1ae318e034f883df819be23f8641

Comment (pyama86) [reply netapp/perf.go:141]: I understand. Thank you for keeping the policy.
I will fix it by myself.

**Pull request merged**